macros/cross_db_utils/width_bucket.sql

```diff
@@ -0,0 +1,59 @@
+{% macro width_bucket(expr, min_value, max_value, num_buckets) %}
+  {{ adapter_macro('dbt_utils.width_bucket', expr, min_value, max_value, num_buckets) }}
+{% endmacro %}
+
+
+{% macro default__width_bucket(expr, min_value, max_value, num_buckets) -%}
+
+    {% set bin_size -%}
+    (( {{ max_value }} - {{ min_value }} ) / {{ num_buckets }} )
+    {%- endset %}
+    (
+        -- to break ties when the amount is eaxtly at the bucket egde
+        case
+            when 
+                mod(
+                    {{ dbt_utils.safe_cast(expr, 'decimal(28,6)') }},
```

Comment:
So, unfortunately `decimal(28, 6)` is not a valid data type on BigQuery. Instead:
```suggestion
                   safe_cast(bin_size as numeric) }}
```
You might have to create a separate `bigquery__width_bucket` macro to implement this logic 😬